### PR TITLE
Channel manifest should be version aware

### PR DESCRIPTION
**Pull request by justinsb**
* We can target AMIs to kubernetes versions
* We can recommend / force a kops upgrade
* We can recommend / force a kubernetes upgrade

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1625)

### CODE REVIEW

File: upup/pkg/fi/cloudup/apply_cluster.go [hunk 1]
```diff
@@ -575,17 +592,12 @@ func findHash(url string) (*hashing.Hash, error) {
 }
 
 // upgradeSpecs ensures that fields are fully populated / defaulted
-func (c *ApplyClusterCmd) upgradeSpecs() error {
+func (c *ApplyClusterCmd) upgradeSpecs(channel *api.Channel) error {
```
Comment (kris-nova) [inline upup/pkg/fi/cloudup/apply_cluster.go]: Should the channel be in the `ApplyClusterCmd` here? Maybe add it as a top level member.. `c.channel` for instance..

File: upup/pkg/fi/cloudup/apply_cluster.go [hunk 2]
```diff
@@ -603,6 +615,117 @@ func (c *ApplyClusterCmd) upgradeSpecs() error {
...
+	required, err := api.IsUpgradeRequired(versionInfo, kopsVersion)
+	if err != nil {
+		glog.Warningf("unable to parse version requirement for kops version %q in channel", kopsVersion)
+	}
+
+	if recommended != "" && !required {
+		fmt.Printf("\n")
+		fmt.Printf("*******************************************************************\n")
+		fmt.Printf("A new kops version is available: %s\n", recommended)
+		fmt.Printf("Upgrading is recommended\n")
```
Comment (kris-nova) [inline upup/pkg/fi/cloudup/apply_cluster.go]: Can we point to something here for reference? `https://github.com/kubernetes/kops/UPGRADE.md` for instance.. so we can have some documentation (that can be easily changed) for the user?

We can just point it to the installation section for day 1 purposes..

Comment (kris-nova) [inline upup/pkg/fi/cloudup/apply_cluster.go]: (This goes for both kops rec/require and k8s rec/require)

File: upup/pkg/fi/cloudup/apply_cluster.go (hunk 1 above)
Comment (justinsb) [inline upup/pkg/fi/cloudup/apply_cluster.go]: Good call

**Pull request merged**